### ISSUE TITLE
Yolo11n-seg in onnx format segmentation breaks

### ISSUE THREAD

**Issue by VohminV** [ultralytics/ultralytics#20723]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

After converting the trained model, the segmentation mask disappears

0: 320x320 1 Y, 18.5ms
Speed: 1.0ms preprocess, 18.5ms inference, 2.0ms postprocess per image at shape (1, 3, 320, 320)

0: 320x320 1 Y, 37.1ms
Speed: 2.0ms preprocess, 37.1ms inference, 16.6ms postprocess per image at shape (1, 3, 320, 320)

### Environment

WARNING ⚠️ torchvision==0.21 is incompatible with torch==2.4.
Run 'pip install torchvision==0.19' to fix torchvision or 'pip install -U torch torchvision' to update both.
For a full compatibility table see https://github.com/pytorch/vision#installation
Ultralytics 8.3.88 🚀 Python-3.12.3 torch-2.4.0+cu121 CPU (Intel Core(TM) i5-6500 3.20GHz)
[W520 13:22:52.371234818 NNPACK.cpp:61] Could not initialize NNPACK! Reason: Unsupported hardware.
YOLO11n-seg summary (fused): 113 layers, 2,834,763 parameters, 0 gradients, 10.2 GFLOPs

PyTorch: starting from 'Y.pt' with input shape (1, 3, 320, 320) BCHW and output shape(s) ((1, 37, 2100), (1, 32, 80, 80)) (5.7 MB)

ONNX: starting export with onnx 1.17.0 opset 19...
ONNX: slimming with onnxslim 0.1.48...
ONNX: export success ✅ 3.5s, saved as 'Y.onnx' (10.9 MB)

Export complete (5.2s)
Results saved to /home/linuxlite
Predict:         yolo predict task=segment model=Y.onnx imgsz=320  
Validate:        yolo val task=segment model=Y.onnx imgsz=320 data=/kaggle/input/helipad/data.yaml  
Visualize:       https://netron.app

### Minimal Reproducible Example

from ultralytics import YOLO

# Load the YOLO11 model
model = YOLO("yolo11n.pt")

# Export the model to ONNX format
model.export(format="onnx")  # creates 'yolo11n.onnx'

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @VohminV, thank you for reporting this issue and for providing detailed information and a minimal reproducible example (MRE) — much appreciated! 🚀 This is an automated response to help you get the fastest possible support. An Ultralytics engineer will review your report and assist you soon.

We encourage you to review the [Docs](https://docs.ultralytics.com/) for guidance on export and segmentation tasks, as well as [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions and troubleshooting tips are covered there.

Since this is a 🐛 Bug Report and you’ve already shared an MRE, you’re on the right track! Please ensure your environment matches our [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) and see our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for more on training and exports.

Join the Ultralytics community to connect with others:
- For real-time chat, join our [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth Q&A, visit [Discourse](https://community.ultralytics.com/)
- For broader discussions, check our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Please ensure you are using the latest version of `ultralytics` and all dependencies in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

If you encounter compatibility warnings (like with torchvision and torch), please follow the guidance in your log or see the full compatibility table in the [PyTorch Vision README](https://github.com/pytorch/vision#installation).

## Environments

You can run YOLO in any of these verified environments (with all dependencies preinstalled):

- **Notebooks** (free GPU): <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests ensure all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) work correctly on macOS, Windows, and Ubuntu.

Thank you for your patience and for helping improve Ultralytics! An engineer will follow up with you soon.

**Comment by Y-T-G** [comment]
Dus you specify `task` while loading the model?

**Comment by VohminV** [comment]
from ultralytics import YOLO

model = YOLO("Y.pt")
model.export(format="onnx", task="segment",  opset=18)

**Comment by Y-T-G** [comment]
You need to specify `task="segment"` when loading the ONNX mdoel.

`model = YOLO("Y.onnx", task="segment")`
